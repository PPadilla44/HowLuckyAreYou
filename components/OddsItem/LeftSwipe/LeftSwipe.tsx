import { StyleSheet, Text, View } from 'react-native'
import React from 'react'
import Colors from '../../../constants/Colors';


const LeftSwipe = () => {
    return (
        <View style={styles.Btn}>
            <Text style={styles.Txt}>
                Share
            </Text>
        </View>
    );
};

export default LeftSwipe

const styles = StyleSheet.create({
    Btn: {
        backgroundColor: Colors.shared.text,
        justifyContent: 'center',
        alignItems: "flex-start"
    },
    Txt: {
        color: 'white',
        fontWeight: '600',
        paddingHorizontal: 30,
        paddingVertical: 20,
    }
})