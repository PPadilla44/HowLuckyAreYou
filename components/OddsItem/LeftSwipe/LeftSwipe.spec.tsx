import { cleanup, fireEvent, render, RenderAPI } from "@testing-library/react-native";
import LeftSwipe from "./LeftSwipe";

describe('<LeftSwipe />', () => {
    

    const mockFn = jest.fn();
    let wrapper: RenderAPI;

    afterEach(cleanup)

    beforeEach(() => {
        wrapper = render(<LeftSwipe callback={mockFn} /> )
    });

    it('should render correctly', () => {
        wrapper.getByTestId("leftSwipeBtn");
        wrapper.getByText("Share");
    });

    it('should run callback on press', () => {
        fireEvent.press(wrapper.getByTestId("leftSwipeBtn"));
        expect(mockFn).toBeCalled()
    });
    
    it('should render correct styling', () => {
        
        expect(wrapper.getByTestId("leftSwipeBtn")).toHaveStyle({
            backgroundColor: "#325C99",
            justifyContent: 'center',
            alignItems: "flex-start"
        });
        expect(wrapper.getByText("Share")).toHaveStyle({
            color: "white",
        })
    });

    it('should match snapshot', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});