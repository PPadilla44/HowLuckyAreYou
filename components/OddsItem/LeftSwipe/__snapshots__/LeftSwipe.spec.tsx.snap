// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftSwipe /> should match snapshot 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  exclusive={true}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#325C99",
      "justifyContent": "center",
    }
  }
  testID="leftSwipeBtn"
>
  <View
    accessible={true}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "600",
          "paddingHorizontal": 30,
          "paddingVertical": 20,
        }
      }
    >
      Share
    </Text>
  </View>
</RNGestureHandlerButton>
`;
